src/createStore.js

```diff
@@ -165,12 +166,35 @@ export default function createStore(reducer, preloadedState, enhancer) {
    * The state still keep changing between dispatches, only listeners (subscribers to store) don't run.
    * Set to true if listeners should be notified only once per frame.
    * Set to false (default) if listeners should be notified after every dispatch
+   * As requestAnimationFrame is disabled while tab is inactive, so in case it's in use,
+   * we'll use the other method when leaving the tab, and returning to it when the tab is active again
    */
   function setNotifyListenersOnNextFrame(shouldNotifyOnNextFrame) {
     isNotifyListenersNextFrame = shouldNotifyOnNextFrame
+
+    if (shouldNotifyOnNextFrame && !hasVisibilityListener) {
+      hasVisibilityListener = true;
```

Comment:
maybe we should adda parameter "shouldRemoveVisibilityListener = true"
and in line 182, you should call setNotifyListenersOnNextFrame(false, false);

so if it is called from outside with (false) you should remove the listener (if exists) since it is not needed